Raiders update

Imienny1 opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->
:cl: Imienny
tweak: Tweaked a bit list of guns which raiders can get.
tweak: Raiders can now find random backpacks in their base.
tweak: Buffed raiders hardsuits.
rscadd: Added bunch of random plate carriers and accessories to raiders ship.
rscadd: Added bag of detergent pods to raiders base, not rated for human consumption.
rscadd: Gave raiders few extra holdout weapons.
rscadd: Raiders once again have trash spawners in their base.
rscadd: Added few new uniforms to raiders uniform list.
/:cl:

Reasons for improving  raiders hardsuits:
raiders were getting really awful hardsuits, most of them was worse than even worst voidsuit, 
and those which werent that bad were always worse than good voidsuit,
i tried to make them more useful by spawning them with modules.

Reason for giving raiders holdout weapons:
they are awful weak against MA and blue tide so they wont affect balance by too much 
and due to their small size they will be very useful for raider who have very limited storage space.

Imienny1 committed:
Raiders update.

Spookerton commented:
Huge blob of personal balance changes. Pass on the hazard suit and a *lot* of the gun related choices tbh.

Tennessee116 commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -298,22 +297,70 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
+
+/obj/item/paper/detergentpods
+	name = "Detergent pods IMPORTANT ! ! !"
+	info = "What are you guys doing with those damn detergent pods, eating them?<BR>our washing mashine is still broken and despite that we used 3 bags in last week!<BR><BR>and someone tell vox to cut use of lavender perfumes, it smells like if it escaped from detergent factory."
```
```suggestion
	info = "What are you guys doing with those damn detergent pods, eating them?<BR>our washing machine is still broken and despite that we used 3 bags in last week!<BR><BR>and someone tell vox to cut use of lavender perfumes, it smells like if it escaped from detergent factory."
```
Typo might be intentional, don't really know.

Imienny1 replied:
Its not, i will fix it soon-ish when i get response from spook

Imienny1 commented:
I replaced half of plate carriers with pirate coats to give option to pirate coat fans to choose between plate carriers and pirate coats.
i also got rid of weapons spawners from trash spawner which i added and removed one of holdout weapons spawner and hazard rigs.

SierraKomodo commented on code/modules/clothing/spacesuits/rig/suits/light.dm:
```diff
@@ -59,6 +59,17 @@
 	glove_type = /obj/item/clothing/gloves/lightrig/hacker
 	boot_type = /obj/item/clothing/shoes/lightrig/hacker
 
+	initial_modules = list(
+		/obj/item/rig_module/ai_container,
+		/obj/item/rig_module/power_sink,
+		/obj/item/rig_module/datajack,
+		/obj/item/rig_module/electrowarfare_suite,
+		/obj/item/rig_module/voice,
+		/obj/item/rig_module/vision,
+		/obj/item/rig_module/cooling_unit
+		)
+
+/obj/item/rig/light/hacker/nvg
```
Why make a separate /nvg alternative and remove nvgs from the original?

Imienny1 replied:
To give raiders their own thermal-less rig without touching loot on away sites.

SierraKomodo replied:
Right well, two things:
1: The hacker rig doesn't appear to be used anywhere else or at away sites
2: The original hacker rig didn't include thermals in the first place, only NVGs

SierraKomodo commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -276,16 +274,16 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
 				/obj/item/gun/projectile/automatic/machine_pistol,
+				/obj/item/gun/projectile/automatic/sec_smg,
 				/obj/item/gun/projectile/pistol/military/alt,
-				/obj/item/gun/projectile/pistol/holdout,
```
I don't really see why this is pulled from the lilgun list but kept in the raider spawn list? Feels inconsistent.

Imienny1 replied:
I dont remember why i left it in raiders gun list but most likely i forgot about it,
i removed it from lilgun spawner because of holdout gun spawner.

SierraKomodo commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -276,16 +274,16 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
 				/obj/item/gun/energy/plasmacutter,
 				/obj/item/gun/energy/incendiary_laser,
 				/obj/item/gun/projectile/automatic/machine_pistol,
+				/obj/item/gun/projectile/automatic/sec_smg,
```
I'm confused why you removed this from the raider spawns then re-added it to the gun lists? Especially to the gun list that's for pistols, not SMGs?

SierraKomodo commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -276,16 +274,16 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
 				/obj/item/gun/energy/stunrevolver,
 				/obj/item/gun/projectile/shotgun/doublebarrel/sawn,
 				/obj/item/gun/energy/xray/pistol,
-				/obj/item/gun/energy/pulse_rifle/pistol,
```
Also unsure why this one's being removed from the lists

Imienny1 replied:
Its very weak, pretty much noob trap.

SierraKomodo commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -298,22 +296,68 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
 				/obj/item/gun/energy/laser,
-				/obj/item/gun/energy/captain,
 				/obj/item/gun/energy/pulse_rifle,
-				/obj/item/gun/energy/pulse_rifle/carbine,
```
Another one that's fully removed and I'm unsure why

Imienny1 replied:
As above, its very weak.

SierraKomodo commented on code/game/antagonist/outsider/raider.dm:
```diff
@@ -298,22 +296,68 @@ GLOBAL_DATUM_INIT(raiders, /datum/antagonist/raider, new)
+				/obj/item/storage/box/monkeycubes/stokcubes = 4,
+				/obj/item/storage/box/monkeycubes/farwacubes = 4,
+				/obj/item/storage/firstaid/surgery = 4,
+				/obj/item/cell/infinite = 1,
```
Very iffy on just giving this to antags as a possible spawn, even if it's a minor chance.

Imienny1 replied:
I didnt noticed it when re-adding this spawner, i will remove it soon.

SierraKomodo commented on code/game/objects/random/random.dm:
```diff
@@ -308,6 +308,18 @@
 				/obj/item/gun/energy/retro = 1,
 				/obj/item/gun/projectile/pistol/throwback = 1)
 
+/obj/random/smallhandgun
+	name = "Random Small Handgun"
+	desc = "This is a random small sidearm."
+	icon = 'icons/obj/guns/holdout_pistol.dmi'
+	icon_state = "pistol"
+
+/obj/random/smallhandgun/spawn_choices()
+	return list(/obj/item/gun/projectile/pistol/holdout = 2,
+				/obj/item/gun/energy/gun/small = 2,
+				/obj/item/gun/projectile/revolver/holdout = 2,
+				/obj/item/storage/box/syndie_kit/silenced = 1)
```
What is the purpose of adding this instead of placing another `lilgun` spawner?

Imienny1 replied:
To guarantee that they will spawn with at least one holdout/stealthy weapon which can be stored in pocket in case if they spawn without holster
and without buffing them too much by giving them spawner which may spawn 15mm handgun.

SierraKomodo commented on code/game/objects/random/random.dm:
```diff
@@ -728,6 +740,26 @@ obj/random/obstruction/spawn_choices()
 
+/obj/random/backpack
+	name = "random backpack"
+	desc = "This is random backpack."
```
```suggestion
	desc = "This is a random backpack."
```

SierraKomodo commented on code/game/objects/random/random.dm:
```diff
@@ -840,6 +872,88 @@ obj/random/obstruction/spawn_choices()
+
+/obj/random/pcarrieraccessory
+	name = "random plate carrier accessory"
+	desc = "This is a random armored plate for plate carrier."
```
The description isn't accurate to the actual list of options.

SierraKomodo commented on maps/torch/torch_antagonism.dm:
```diff
@@ -60,3 +60,30 @@
+		/obj/item/clothing/under/det,
+		/obj/item/clothing/under/color/brown,
+		/obj/item/clothing/under/rank/security2,
+		/obj/item/clothing/under/solgov/utility,
+		/obj/item/clothing/under/solgov/utility/fleet
```
Not interested in giving crew uniforms to off-ship roles.

SierraKomodo commented on maps/torch/torch_antagonism.dm:
```diff
@@ -60,3 +60,30 @@
+		/obj/item/clothing/suit/storage/toggle/hoodie/black,
+		/obj/item/clothing/suit/unathi/mantle,
+		/obj/item/clothing/suit/poncho/colored,
+		/obj/item/clothing/suit/storage/solgov/service,
+		/obj/item/clothing/suit/storage/solgov/service/fleet/officer
```
Not interested in giving crew uniforms to off-ship roles.

SierraKomodo reviewed (changes requested):
A lot of changes here feel arbitrary or don't make sense to me. See individual line reviews.

The armor I'm also hesitant of - Tactical armor is effectively merc armor when facing the Torch, due to a lack of ballistic weaponry.

Overall, I'm not sure what, if anything, this PR really adds to raider outside of making them closer to merc alternative (And at that point why does raider even exist in the first place).

Imienny1 commented:
> A lot of changes here feel arbitrary or don't make sense to me. See individual line reviews.
> 
> The armor I'm also hesitant of - Tactical armor is effectively merc armor when facing the Torch, due to a lack of ballistic weaponry.
> 
> Overall, I'm not sure what, if anything, this PR really adds to raider outside of making them closer to merc alternative (And at that point why does raider even exist in the first place).

Its a bit old PR so i forgot why i did some of these changes.
There is no need for ballistics, tac armor protects only only against e-guns and it has no real protection against laser rifles
so its on pair with blood red voidsuits(which raiders already get) but without limb protection and worse stun resistance, 

Main point of this PR is to make them look more like ragtag pirates by giving them more randomized equipment
and to do low effort update of their equipment to make them less likely to spawn with weak guns.

SierraKomodo commented:
Right, after further consideration and discussion internally we're not really interested in the changes being made here at all, so instead of keeping this going on life support, I'm cutting it here.
